Casmpet 5341 1.0 addremovereplace ncn (#1208)
Cray-HPE/docs-csm: operations/node_management/Add_Remove_Replace_NCNs/Boot_NCN.md

### Step 2 - Set the PXE boot option and power on the node

> **IMPORTANT:** These commands assumes you have set the variables from [the prerequisites section](../Add_Remove_Replace_NCNs.md#add-ncn-prerequisites).

Reviewer: ```suggestion
**IMPORTANT:** These commands assume the variables from [the prerequisites section](../Add_Remove_Replace_NCNs.md#add-ncn-prerequisites) have been set.
```

1. Wait until `cloud-init` displays messages similar to these on the console to indicate that cloud-init has finished with the module called `modules:final`.

    ```screen
    [  300.390000] cloud-init[7110]: 2022-03-16 18:30:59,449 - util.py[DEBUG]: cloud-init mode 'modules' took 244.143 seconds (198.87)
    [  300.390106] cloud-init[7110]: 2022-03-16 18:30:59,449 - handlers.py[DEBUG]: finish: modules-final: SUCCESS: running modules for final
    [  OK  ] Started Execute cloud user/final scripts.
    [  OK  ] Reached target Cloud-init target.
    ```

1. Then press enter on the console to ensure that the the login prompt is displayed including the correct hostname of this node. Then exit the ConMan console (**&** then **.**), and then use `ssh` to log in to the node to complete any remaining steps based on the node type.

> ### Step 4 - Set the wipe flag to safeguard against the disk being wiped when the node is rebooted.

Reviewer: remove the period at the end of this line

    [[results.layers]]
    cloneUrl = "https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git"
    commit = "1dc4038615cebcfad3e8230caecc885d987e8148"
    name = "csm-ncn-1.6.28"
    playbook = "site.yml"
    ```
    
2. Or determine the configuration applied another NCN of the same type. This example checks the configuration on `ncn-w002`.

    ```bash
    ncn-mw# cray cfs components describe $(ssh ncn-w002 cat /etc/cray/xname)
    ```

    Example Output:

    ```screen
    configurationStatus = "configured"
    desiredConfig = "ncn-personalization"
    enabled = true
    errorCount = 0
    id = "x3000c0s9b0n0"
    [[state]]
    cloneUrl = "https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git"
    commit = "1dc4038615cebcfad3e8230caecc885d987e8148"
    lastUpdated = "2022-03-15T15:29:20Z"
    playbook = "site.yml"
    sessionName = "batcher-5e431205-a4b4-4a2e-8be3-21cf058774cc"
    ```

3. Select the appropriate configuration based on the above steps to personalize the added NCN. In this example, the `ncn-personalization` configuration is used.

    ```bash
    ncn-mw# cray cfs components update $XNAME --desired-config ncn-personalization
    ```

> 4. Wait for `configurationStatus` to transistion from `pending` to `configured`

Reviewer: missing period

    ```bash
    ncn-mw# watch "cray cfs components describe $XNAME"
    ```

    Example Output:

    ```
    configurationStatus = "configured"
    desiredConfig = "ncn-personalization"
    ...
    ```

### Step 7 - Lock the management nodes

> Follow [How to Lock Management Single Node](../../../operations/hardware_state_manager/Lock_and_Unlock_Management_Nodes.md#to-lock-single-nodes-or-lists-of-specific-nodes-and-their-bmcs). The management nodes may be unlocked at this point. Locking the management nodes and their BMCs will prevent actions from FAS to update their firmware or CAPMC to power off or do a power reset. Doing any of these by accident will take down a management node. If the management node is a Kubernetes master or worker node, this can have serious negative effects on system operation.

Reviewer: ```suggestion
Follow the [How to Lock Management Single Node](../../../operations/hardware_state_manager/Lock_and_Unlock_Management_Nodes.md#to-lock-single-nodes-or-lists-of-specific-nodes-and-their-bmcs) procedure. The management nodes may be unlocked at this point. Locking the management nodes and their BMCs will prevent actions from FAS to update their firmware or CAPMC to power off or do a power reset. Doing any of these by accident will take down a management node. If the management node is a Kubernetes master or worker node, this can have serious negative effects on system operation.
```

### Step 8 - **For Storage nodes only**

Follow [Add Ceph Node](../../utility_storage/Add_Ceph_Node.md) to join the added storage node to the Ceph cluster.

### Step 9 - **For ncn-m001 only**

> 1. Restore and verify the site link for ncn-m001.

Reviewer: ```suggestion
1. Restore and verify the site link for `ncn-m001`.
```

>     Use the $CAN_IP that was recorded prior to powering down ncn-m001 to access ncn-m002.

Reviewer: ```suggestion
    Use the $CAN_IP that was recorded prior to powering down `ncn-m001` to access `ncn-m002`.
```
